go/test/endtoend/orchestrator/orc_test.go

```diff
@@ -0,0 +1,361 @@
+)
+
+var (
+	// ClusterInstance instance to be used for test with different params
+	clusterInstance *cluster.LocalProcessCluster
+	keyspaceName    = "ks"
+	shardName       = "0"
+	keyspace        *cluster.Keyspace
+	shard0          *cluster.Shard
+	//dbName          = "vt_" + keyspaceName
+	//username        = "vt_dba"
+	hostname = "localhost"
+	cell1    = "zone1"
+	cell2    = "zone2"
+	//	insertSQL       = "insert into vt_insert_test(id, msg) values (%d, 'test %d')"
+	//	sqlSchema       = `
+	//	create table vt_insert_test (
+	//	id bigint,
+	//	msg varchar(64),
+	//	primary key (id)
+	//	) Engine=InnoDB
+	//	`
+	tablets []*cluster.Vttablet
+)
+
+func createCluster(t *testing.T, numReplicas int, numRdonly int) (*cluster.LocalProcessCluster, int) {
+	clusterInstance = cluster.NewCluster(cell1, hostname)
+
+	// Launch keyspace
+
+	// Start topo server
+	err := clusterInstance.StartTopo()
+	if err != nil {
+		return nil, 1
+	}
+
+	// Adding another cell in the same cluster
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+cell2)
+	if err != nil {
+		return nil, 1
+	}
+	err = clusterInstance.VtctlProcess.AddCellInfo(cell2)
+	if err != nil {
+		return nil, 1
+	}
+
+	uidBase := 100
+	for i := 0; i < numReplicas; i++ {
+		tablets = append(tablets, clusterInstance.NewVttabletInstance("replica", uidBase+i, cell1))
+	}
+	for i := 0; i < numRdonly; i++ {
+		tablets = append(tablets, clusterInstance.NewVttabletInstance("rdonly", uidBase+numReplicas+i, cell1))
+	}
+
+	clusterInstance.VtTabletExtraArgs = []string{
+		"-lock_tables_timeout", "5s",
+		"-disable_active_reparents",
+	}
+
+	// Initialize Cluster
+	err = clusterInstance.SetupCluster(keyspace, []cluster.Shard{*shard0})
+	if err != nil {
+		return nil, 1
+	}
+
+	// OK to do this since we know we have only one keyspace and shard
+	keyspace = &clusterInstance.Keyspaces[0]
+	shard0 = &keyspace.Shards[0]
+	shard0.Vttablets = tablets
+
+	//Start MySql
+	var mysqlCtlProcessList []*exec.Cmd
+	for _, tablet := range shard0.Vttablets {
+		log.Infof("Starting MySql for tablet %v", tablet.Alias)
+		proc, err := tablet.MysqlctlProcess.StartProcess()
+		if err != nil {
+			return nil, 1
+		}
+		mysqlCtlProcessList = append(mysqlCtlProcessList, proc)
+	}
+
+	// Wait for mysql processes to start
+	for _, proc := range mysqlCtlProcessList {
+		if err := proc.Wait(); err != nil {
+			return nil, 1
+		}
+	}
+
+	return clusterInstance, 0
+
+}
+
+// Cases to test:
+// 1. create cluster with 1 replica and 1 rdonly, let orc choose master
+// verify rdonly is not elected, only replica
+// verify replication is setup
+func TestMasterElection(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	clusterInstance, exitCode := createCluster(t, 1, 1)
+	if exitCode != 0 {
+		os.Exit(exitCode)
+	}
+	defer func() {
+		clusterInstance.Teardown()
+		tablets = []*cluster.Vttablet{}
+	}()
+	for _, tablet := range tablets {
+		// Create Database
+		err := tablet.VttabletProcess.CreateDB(keyspaceName)
```

Comment:
This shouldn't be needed any more. Starting the vttablet should create the db automatically. Also, I'm surprised this worked because the replicas will all have diverging gtids.

Reply:
I'm not sure why this is _NOT_ creating problems, but I will remove the call to CreateDB.